src/NLog/Layouts/Typed/Layout.cs

```diff
@@ -0,0 +1,315 @@
+                if (e.MustBeRethrown())
+                {
+                    throw;
+                }
+            }
+
+            value = default(T);
+            return false;
+        }
+
+        #endregion
+
+        #region Conversion
+
+        /// <summary>
+        /// Converts a given text to a <see cref="Layout" />.
+        /// </summary>
+        /// <param name="value">Text to be converted.</param>
+        /// <returns><see cref="SimpleLayout" /> object represented by the text.</returns>
+        public static implicit operator Layout<T>(T value)
+        {
+            return new Layout<T>(value);
+        }
+
+        /// <summary>
+        /// Converts a given text to a <see cref="Layout" />.
+        /// </summary>
+        /// <param name="layout">Text to be converted.</param>
+        /// <returns><see cref="SimpleLayout" /> object represented by the text.</returns>
+        public static implicit operator Layout<T>([Localizable(false)] string layout)
+        {
+            return new Layout<T>(layout);
+        }
+
+        #endregion
+
+        /// <inheritdoc />
+        protected override string GetFormattedMessage(LogEventInfo logEvent)
+        {
+            if (_fixedValue)
+            {
+                if (_value == null)
+                {
+                    return null;
+                }
+
+                var text = ValueToString(_value, LoggingConfiguration.DefaultCultureInfo);
+                if (text != null)
+                {
+                    return text;
+                }
+            }
+
+            return _layout.Render(logEvent);
+        }
+
+        /// <inheritdoc cref="IRawValue" />
+        internal override bool TryGetRawValue(LogEventInfo logEvent, out object rawValue)
+        {
+            if (_fixedValue)
+            {
+                rawValue = _value;
+                return true;
+            }
+
+            if (_layout == null)
+            {
+                rawValue = null;
+                return true;
+            }
+
+            if (_layout.TryGetRawValue(logEvent, out var raw))
+            {
+                var success = TryConvertRawToValue(raw, out var i);
+                rawValue = i;
+                return success;
+            }
+
+            rawValue = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Render to value
+        /// </summary>
+        /// <returns></returns>
+        T IRenderable<T>.RenderToValue(LogEventInfo logEvent)
+        {
+            return RenderToValueInternal(logEvent, null);
+        }
+
+        /// <summary>
+        /// Render to value
+        /// </summary>
+        /// <param name="logEvent"></param>
+        /// <param name="reusableBuilder">if null, default layout render will be used</param>
+        /// <returns></returns>
+        internal T RenderToValueInternal(LogEventInfo logEvent, [CanBeNull] StringBuilder reusableBuilder)
+        {
+            if (_fixedValue)
```

Comment:
Duplicate code with `TryGetRawValue`. Maybe call `TryGetRawValue` from this method, and only when that fails, then fallback to rendering.

Reply:
Fixed,we needed an extra method because of T vs object